opengrok-web/src/main/java/org/opengrok/web/PageConfig.java

```diff
@@ -494,6 +477,24 @@ public String canProcess() {
         return dirFileList;
     }
 
+    List<String> getSortedFiles(File[] files) {
+        if (getEnv().getListDirsFirst()) {
+            Arrays.sort(files, (f1, f2) -> {
+                if (f1.isDirectory() && !f2.isDirectory()) {
+                    return -1;
+                } else if (!f1.isDirectory() && f2.isDirectory()) {
+                    return 1;
+                } else {
+                    return f1.getName().compareTo(f2.getName());
+                }
+            });
+        } else {
+            Arrays.sort(files, Comparator.comparing(File::getName));
+
+        }
+        return Arrays.stream(files).map(File::getName).collect(Collectors.toList());
```

Comment:
What about going one step further? In this stream we could use something like `sorted(getFileComparator())`.

Reply:
nice idea !